pkg/cache/tas_flavor_snapshot.go

```diff
@@ -809,22 +921,48 @@ func (s *TASFlavorSnapshot) lowerLevelDomains(domains []*domain) []*domain {
 
 func (s *TASFlavorSnapshot) sortedDomains(domains []*domain, unconstrained bool) []*domain {
 	result := slices.Clone(domains)
-	slices.SortFunc(result, func(a, b *domain) int {
+
+	switch {
+	case useBestFitAlgorithm(unconstrained):
+		slices.SortFunc(result, sortDomainsForMostFreeCapacity)
+	case useLeastFreeCapacityAlgorithm(unconstrained):
+		slices.SortFunc(result, sortDomainsForLeastFreeCapacity)
+	default:
+		slices.SortFunc(result, sortDomainsForMostFreeCapacity)
+	}
+
+	return result
+}
+
+func sortDomainsForMostFreeCapacity(a, b *domain) int {
+	if a.chunkState == b.chunkState {
```

Comment:
here and elsewhere: please flatten

```suggestion
	if a.chunkState != b.chunkState {
	  return
	}
```